### ISSUE TITLE
Discount Mode

### ISSUE THREAD

**Issue by Iwan1803** [Mmtrx/FS22_BetterContracts#58]
Regarding to #39:
Is it possible to display the achieved purchase discount for fields somewhere?
So a display where you can see the original price and the one with discount deducted?

**Comment by Mmtrx** [comment]
Hi Iwan,
the current discount for a farmland is shown, when you select it on the ingame map in "farmland" (X) mode. Also in the Buy dialog box (if you hit spacebar):
![discount](https://user-images.githubusercontent.com/86842734/200129382-0014b75e-2d85-494f-b83b-1d0f980aae3b.png)
![showDiscount](https://user-images.githubusercontent.com/86842734/200129384-7dfb0b25-4089-4a91-a05a-c898a2e78233.png)

**Comment by Iwan1803** [comment]
Ah, ok, danke 👍 
I thought that with everyt completed order a discount will be already calculated. However, I do not see any or am I wrong?
Maybe my entry in FS_BetterContracts.xml is wrong:
<FS22_BetterContracts debug="true" maxActive="0" lazyNPC="true" discountMode="true">

**Comment by Iwan1803** [comment]
I deleted FS_BetterContracts.xml in modSettings folder and the parameter was wrong:
_<FS22_BetterContracts debug="true" maxActive="0" lazyNPC="true" hard="false" discount="true">_
Still can't see any discount and thought, the mod "FS22_additionalFieldInfo" could cause this, but after deactivation still can't see any discount on any field.
Any idea?

**Comment by Mmtrx** [comment]
Hi,
the intended logic goes like this:
- complete a contract on a field 
- in the ingame map overview click X to display farmlands 
- click on the field/farmland you did the contract on 
- you should see the discount 

If this doesn‘t work, pls send your log.txt. 
Is this on an MP Game? Discount mode is not yet tested on MP.

**Comment by Iwan1803** [comment]
In SP i completed contracts on 2 different fileds and checked the fields afterwards, but there was nothing shown.
In the log no error or warning was shown and yes, debug was activated.
`2022-11-06 16:01   Info: FS22_BetterContracts read config: maxActive 0, lazyNPC, discountMode`
I deactivated then "FS22_additionalFieldInfo", recovered the savegame, completed the 2 contracts but again no discount was shown. That was on the map "Landkreis Rostock".
I tried it then on a 2nd savegame on the map "Volksleron" and there it worked directly. I'll try it again on Rostock with a fresh contract and will post then an update.

**Comment by Iwan1803** [comment]
I had to skip to the next day as i didn't got any orders, but then it worked - also on a dedi server!

**Comment by Iwan1803** [comment]
Yesterday on my dedi server we've completed an order but no discount was shown.
Should i activate the debug mode to get some messages?

**Comment by Mmtrx** [comment]
Yes please activate debug, and upload your log.txt.

**Comment by Iwan1803** [comment]
Ok, i'll activated it and will come back after completing the next contracts.

**Comment by Iwan1803** [comment]
Sorry for the delay, but i watched for any entry in the server log, but couldn't find any helpful.
The only error which sometimes comes up:
`2022-11-28 20:01 Error: Running LUA method 'packetReceived'.
dataS/scripts/fieldJobs/MissionManager.lua(554) : attempt to index local 'mission' (a nil value)`

Also we are in January and don't get only a very small amount contracts. Most disappear right after the refresh and also sometimes the screen shows no contracts but the button to accept one.

**Comment by Mmtrx** [comment]
Thanks for Testing- v1.2.5.0 is Not MP-Ready Yet . 
I plan to have an early version of 1.2.6.0 up here within a week.

**Comment by Iwan1803** [comment]
Sounds good, i'll wait for it 👍

**Comment by Mmtrx** [comment]
It's here: [v1.2.6.1](https://github.com/Mmtrx/FS22_BetterContracts/releases/tag/v1.2.6.1). 

My humble MP tests were ok, but I am interested in your experience,

**Comment by Iwan1803** [comment]
We are having 1.2.6.0 on our server and it looks like it's working. Have to wait for the next day to see, if all contracts will then be shown.
I'll update to 1.2.6.1 on Wednesday if there is no need to update today.

**Comment by Mmtrx** [comment]
1.2.6.1 only has more (Russian, Brazil) translation. And a bug fix only for Russian (and other utf8 languages). -> No need to update 😉

**Comment by Iwan1803** [comment]
It works not on Dedi server. It shows the deducted price on the PDA but then you have to pay to original price.

**Comment by Mmtrx** [comment]
I could not reproduce that on my test dedi server.
Do you have screen shots / log.txt about that situation?

**Comment by Iwan1803** [comment]
There was nothing logged, but i've got screenshots:
![image](https://user-images.githubusercontent.com/47528836/209565775-ec35cf57-d55e-420e-8118-c15e6e4cf348.png)
![image](https://user-images.githubusercontent.com/47528836/209565871-a95a2732-2503-4ff1-b7f3-e288945b3446.png)
![image](https://user-images.githubusercontent.com/47528836/209565897-e0c1b32e-adf3-4502-8e8a-9d97e4f2e392.png)

**Comment by Mmtrx** [comment]
Do you happen to have mod Dynamic Field Prices active? That‘s Not (yet ) compatible. See also #83

**Comment by Iwan1803** [comment]
No, there is no other mod on the server for field prices, but we are using this one to show additional infos as you can see in the screenshot ("Fläche"):
https://www.farming-simulator.com/mod.php?lang=de&country=de&mod_id=229672&title=fs2022

**Comment by Mmtrx** [comment]
No, the Additional Field Info is not a problem.
Are you using latest modhub version 1.2.6.2?

**Comment by Iwan1803** [comment]
![image](https://user-images.githubusercontent.com/47528836/209680282-289fb3c1-d60d-4aa8-91bc-b887b94d968e.png)

**Comment by Ben040Zin** [comment]
same here on local singleplayer. Tested without any mods but BC 1.2.6.2 
[log.txt](https://github.com/Mmtrx/FS22_BetterContracts/files/10313334/log.txt)

**Comment by Ben040Zin** [comment]
nope

**Comment by Iwan1803** [comment]
Joined our server some minutes ago and activated debug mode, but nothing was written in the client nor the server log:
![image](https://user-images.githubusercontent.com/47528836/209858007-d6f3562d-7b7e-4618-8ac2-2f5fa18885e6.png)
![image](https://user-images.githubusercontent.com/47528836/209858043-b17c452e-6409-4805-8732-0c9ec6561307.png)

**Comment by tkralicek** [comment]
Hi,
I have the same issue in single player. I have found out that the `onClickBuyFarmland` function is never called. 

I have no experience with FS scripting but it seems it is possible to fix this issue by adding the following assignment:

```diff
--- ./betterContracts.lua
+++ ./betterContracts.lua
@@ -567,6 +567,7 @@ function BetterContracts:onPostLoadMap(mapNode, mapFile)
     self.frCon = self.gameMenu.pageContracts
     self.frMap = self.gameMenu.pageMapOverview
     self.frMap.ingameMap.onClickMapCallback = self.frMap.onClickMap
+    self.frMap.buttonBuyFarmland.onClickCallback = self.frMap.onClickBuyFarmland
 
     initGui(self) 			-- setup my gui additions
     self.initialized = true
```

However I don't know if this is a correct way.

**Comment by Mmtrx** [comment]
Thanks, that was a good hint. Problem is that the mod only intercepts the keyboard (space bar) entry to "Buy Land". Mouse is ignored (i.e. does the std thing, w/o discounted price).

Will fix that soon. Meanwhile you could try without the mouse ..

**Comment by Iwan1803** [comment]
Checked it with mouse and space bar on my server: with mouse i should pay the whole price, whit space bar the deducted one.
So there is a workaround till the fix comes 😎

**Comment by Mmtrx** [comment]
There was an issue with the mod not intercepting the mouse-click on the Buy-Land button. Is fixed with [v1.2.6.3](https://github.com/Mmtrx/FS22_BetterContracts/releases/tag/v1.6.2.3)

**Comment by Mmtrx** [comment]
seems ok now
